src/ArduinoCloudThing.cpp

```diff
@@ -460,8 +497,8 @@ ArduinoCloudThing::MapParserState ArduinoCloudThing::handle_LeaveMap(CborValue *
 bool ArduinoCloudThing::ifNumericConvertToDouble(CborValue * value_iter, double * numeric_val) {
 
   if(cbor_value_is_integer(value_iter)) {
-    int val = 0;
-    if(cbor_value_get_int(value_iter, &val) == CborNoError) {
+    uint64_t val = 0;
```

Comment:
I'm not sure that this is correct. If we retrieve an **unsigned int** instead of a **signed int** we are loosing the sign extension. As a consequence **no negative property values would be possible**.

I'd be great if you could write up some decode unit tests where CBOR streams containing negative values are passed. As of now none do exist and therefore this error would have gone unnoticed by the test suite.

Reply:
Correct, I fixed it by declaring a correct type and corresponding cobra library function call. I also added a specific test for negative case. The reason I change the type from int to int64 is that there are cases in which a 32 bit integers doesn't fit the values to manage.